### PR TITLE
Add media demo

### PR DESCRIPTION
**Pull request by yrezgui**

### CODE REVIEW

File: ScopedStorage/app/src/main/java/com/samples/storage/mediastore/AddMediaFragment.kt [hunk 1]
```diff
@@ -16,24 +16,51 @@
...
 
 class AddMediaFragment : Fragment() {
-    private var _binding: FragmentDemoBinding? = null
+    private var _binding: FragmentAddMediaBinding? = null
     private val binding get() = _binding!!
+    private val viewModel: AddMediaViewModel by viewModels()
+
+    private val actionTakePicture = registerForActivityResult(TakePicture()) { success ->
+        if (!success) {
+            Log.d(tag, "Image taken FAIL: ${viewModel.temporaryMediaUri.value}")
```
Comment (nic0lette) [inline ScopedStorage/app/src/main/java/com/samples/storage/mediastore/AddMediaFragment.kt:42]: nit: maybe make `tag` a `const val TAG`?

File: ScopedStorage/app/src/main/java/com/samples/storage/mediastore/AddMediaViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        } else {
+            MediaStore.Images.Media.EXTERNAL_CONTENT_URI
+        }
+
+        val newImage = ContentValues().apply {
+            put(MediaStore.Images.Media.DISPLAY_NAME, generateFilename(source, "jpg"))
+        }
+
+        val uri = context.contentResolver.insert(imageCollection, newImage)
+        _temporaryMediaUri.value =uri
```
Comment (nic0lette) [inline ScopedStorage/app/src/main/java/com/samples/storage/mediastore/AddMediaViewModel.kt:55]: nit: formatting (might make sense to just auto format?)

File: ScopedStorage/app/src/main/java/com/samples/storage/mediastore/AddMediaViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    private val _currentMedia: MutableStateFlow<Media?> = MutableStateFlow(null)
+    val currentMedia: StateFlow<Media?> = _currentMedia
+
+    fun loadMedia() {
+        // Once we get a result from [TakePicture] and [TakeVideo], we clear the value of the
+        // temporaryMediaUri property
+        _temporaryMediaUri.value = null
+    }
+
+    // [TakePicture] and [TakeVideo] activityResult actions isn't returning the URI once it's
```
Comment (nic0lette) [inline ScopedStorage/app/src/main/java/com/samples/storage/mediastore/AddMediaViewModel.kt:36]: nit: if you're using the link style for [TakePicture], etc... then make this a KDOC comment?

File: ScopedStorage/app/src/main/res/layout/fragment_add_media.xml [hunk 4]
```diff
@@ -0,0 +1,111 @@
+<?xml version="1.0" encoding="utf-8"?><!--
```
Comment (nic0lette) [inline ScopedStorage/app/src/main/res/layout/fragment_add_media.xml:1]: nit: <!-- on the next line

**Review by nic0lette — APPROVED** [review]
Just a few nits otherwise lgtm

File: ScopedStorage/app/src/main/java/com/samples/storage/mediastore/AddMediaFragment.kt (hunk 1 above)
Comment (yrezgui) [reply ScopedStorage/app/src/main/java/com/samples/storage/mediastore/AddMediaFragment.kt:36]: Why? It's a property from Fragment (which avoids having to write a constant)

Comment (nic0lette) [reply ScopedStorage/app/src/main/java/com/samples/storage/mediastore/AddMediaFragment.kt:36]: Ah, I didn't notice that. sgtm then.

**Pull request merged**